src/openforms/registrations/contrib/objects_api/helpers.py

```diff
@@ -0,0 +1,39 @@
+from openforms.contrib.zgw.clients.utils import get_today
+from openforms.submissions.mapping import apply_data_mapping
+from openforms.submissions.models import Submission
+
+from ...constants import REGISTRATION_ATTRIBUTE
+from .client import get_objects_client
+from .utils import render_to_json
+
+
+def prepare_data_for_registration(
+    submission: Submission, context: dict, options: dict, object_mapping: dict
+) -> dict:
+    # Prepare the submission data for sending it to the Objects API. This requires
+    # rendering the configured JSON template and running some basic checks for
+    # security and validity, before throwing it over the fence to the Objects API.
+
+    record_data = render_to_json(options["content_json"], context)
+
+    object_data = {
+        "type": options["objecttype"],
+        "record": {
+            "typeVersion": options["objecttype_version"],
+            "data": record_data,
+            "startAt": get_today(),
+        },
+    }
+    apply_data_mapping(submission, object_mapping, REGISTRATION_ATTRIBUTE, object_data)
+
+    return object_data
+
+
+def send_data_to_api(object_data: dict) -> dict:
+    with get_objects_client() as objects_client:
+        response = objects_client.post(
+            "objects", json=object_data, headers={"Content-Crs": "EPSG:4326"}
+        )
+        response.raise_for_status()
+
+    return response.json()
```

Comment:
I think it's neater to define a client subclass in `openforms.contrib.objects_api.client` which is returned by `get_objects_client()` so that you can then call:

```py
client.create_object(object_data)
```
& possibly even break out some parameters into their own arguments so you can enforce that all the necessary information is provided, like the object type URL + version. The peculiarities of the record/object data/shape are then encapsulated on the client, and the objects API/ZGW backend don't need to bother with that, they can just pass along the record data and not worry about other things.